tcms/testplans/static/testplans/js/mutable.js

```diff
@@ -0,0 +1,20 @@
+$(document).ready((function () {
+    populateProductVersion();
+
+    $('#id_product').change(populateProductVersion);
+
+    $('#add_id_product').click(function () {
+        return showRelatedObjectPopup(this);
+    });
+
+    $('#add_id_version').click(function () {
+        return showRelatedObjectPopup(this);
+    });
+}));
+
+function populateProductVersion() {
+    $('#id_version').find('option').remove();
+    $('#add_id_version').get(0).href += `&product=${$('#id_product').val()}`;
```

Comment:
This will not work as expected. Try selecting a different Product a few times without changing the Version. The new parameter will be appended to the URL, not replaced. You have to make sure there is only 1 `product=X` parameter to the URL.




Also `get(0)` I more often written as `$('selector')[0]` to get the first element.

Reply:
Again, good catch. I fixed it with a dirty hack.